includes/Integrations/Jetpack.php

```diff
@@ -68,6 +83,114 @@ public function add_to_jetpack_sitemap( $post_types ) {
 		return $post_types;
 	}
 
+	/**
+	 * Add VideoPress to allowed mime types. If the site does not support videopress, this allow will be filtered out.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param array $mime_types Associative array of allowed mime types per media type (image, audio, video).
+	 *
+	 * @return array
+	 */
+	public function add_videopress( array $mime_types ) {
+		$mime_types['video'][] = self::VIDEOPRESS_MIME_TYPE;
+
+		return $mime_types;
+	}
+
+	/**
+	 * Filter admin ajax responses to change video/videopress back to mp4.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param array   $response   Array of prepared attachment data. @see wp_prepare_attachment_for_js().
+	 * @param WP_Post $attachment Attachment object.
+	 *
+	 * @return array
+	 */
+	public function filter_admin_ajax_response( array $response, WP_Post $attachment ) {
+		if ( self::VIDEOPRESS_MIME_TYPE !== $attachment->post_mime_type ) {
+			return $response;
+		}
+
+		// Get data from post, so we filter out requests from other locations.
+		$post_data = $_POST; // phpcs:ignore WordPress.Security.NonceVerification.Missing
+		if ( ! isset( $post_data['action'] ) || 'query-attachments' !== $post_data['action'] ) {
+			return $response;
+		}
+
+		if ( ! isset( $post_data['query']['source'] ) || 'web_stories_editor' !== $post_data['query']['source'] ) {
+			return $response;
+		}
+
+		// Reset mime type back to mp4, as this is the correct value.
+		$response['mime'] = 'video/mp4';
+		// Make video as optimized.
+		$response['media_source'] = 'video-optimization';
+
+		return $response;
+	}
+
+	/**
+	 * Filter REST API responses to change video/videopress back to mp4.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param WP_REST_Response $response The response object.
+	 * @param WP_Post          $post     The original attachment post.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function filter_api_response( WP_REST_Response $response, WP_Post $post ) {
+		if ( self::VIDEOPRESS_MIME_TYPE !== $post->post_mime_type ) {
+			return $response;
+		}
+
+		$data = $response->get_data();
+
+		// Reset mime type back to mp4, as this is the correct value.
+		$data['mime_type'] = 'video/mp4';
+		// Make video as optimized.
+		$data['media_source'] = 'video-optimization';
+
+		if ( isset( $data['media_details']['videopress']['duration'] ) ) {
+			$data['media_details']['length_formatted'] = $this->format_milliseconds( $data['media_details']['videopress']['duration'] );
+		}
+
+		$response->set_data( $data );
+
+		return $response;
+	}
+
+	/**
+	 * Format milliseconds into seconds.
+	 *
+	 * @since 1.7.2
+	 *
+	 * @param int $milliseconds Milliseconds
+	 *
+	 * @return string
+	 */
+	protected function format_milliseconds( $milliseconds ) {
```

Comment:
Does WP perhaps offer a function for this?

Reply:
I did look for one, I didn't think anything that outputted something useful. The closest that we came to is [human_time_diff](https://github.com/WordPress/wordpress-develop/blob/2382765afa36e10bf3c74420024ad4e85763a47c/src/wp-includes/formatting.php#L3726). If there is something that I am missing, let me know.

Reply:
Just felt like something WP would have a function for, but seems like there indeed isn't a super nice one.

In core, the `length_formatted` formatted string is coming from `getid3_lib::PlaytimeString()`, a static method which accepts seconds and returns `H:M:S` as a string.